### PR TITLE
PRO-475 Tokens Pane

### PR DESCRIPTION
**Pull request by hemadri-tank-bacancy**
* Implemented two tabs tokens and stables
* Normal tokens list sorted by USD
* Added "more bar" button

### CODE REVIEW

File: src/components/Wallet/WalletSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        symbol={item.symbol}
+        name={item.name}
+        iconUrl={item.iconUrl}
+        balance={wrapBigNumberOrNil(item.balance?.balance)}
+        style={{ paddingLeft: 0, paddingRight: 0 }}
+        onPress={() => {
+          onNavigate(item, item.chain);
+          // const assetData = buildAssetDataNavigationParam(item, item.chain);
+          // assetData.backDashboad = true;
+          // navigation.navigate(ASSET, { assetData });
```
Comment (IAmKio) [inline src/components/Wallet/WalletSelection.tsx:97]: Could we remove dead code please and inline the `onNavigate` function call

File: src/components/Wallet/WalletSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+  const colors = useThemeColors();
+  let listOfAssets = tabIndex === 0 ? nonStableTokens : tokens;
+
+  const items = [
+    { key: TOKENS, title: TOKENS, component: null, color: colors.primaryAccent250 },
+    { key: STABLES, title: STABLES, component: null, color: colors.synthetic180 },
+  ];
+
+  const renderItem = (item: any) => {
```
Comment (IAmKio) [inline src/components/Wallet/WalletSelection.tsx:83]: Is there an interface somewhere for `item`?

File: src/components/layout/ButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  items: any[];
+  tabIndex: number;
+  onTabIndexChange: (index: number) => void;
+};
+
+export default function ({ items, tabIndex, onTabIndexChange }: Props) {
+  const colors = useThemeColors();
+
+  const renderItem = (item, index) => {
+    const width = (Dimensions.get('screen').width - spacing.large * 2) / items?.length;
```
Comment (IAmKio) [inline src/components/layout/ButtonGroup.tsx:41]: Will this line calculate a result if `items` is `null`?

File: src/hooks/assets.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const assets: any = useFromAssets();
+
+  if (!listOfStableToken || !assets?.[0])
+    return {
+      tokens: [],
+      percentage: 0,
+    };
+
+  const listOfSymbol = listOfStableToken.map((token) => token.symbol);
+  const listOfAddress = listOfStableToken.map((token) => token.address);
```
Comment (IAmKio) [inline src/hooks/assets.ts:38]:
```suggestion
  const listOfSymbols = listOfStableToken.map((token) => token.symbol);
  const listOfAddresses = listOfStableToken.map((token) => token.address);
```

File: src/hooks/assets.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  if (!stableToken && !assets?.[0])
+    return {
+      tokens: [],
+      percentage: 0,
+      totalPercentage: 0,
+    };
+
+  const listOfSymbol = stableToken.map((token) => token.symbol) ?? [];
+  const listOfAddress = stableToken.map((token) => token.address) ?? [];
```
Comment (IAmKio) [inline src/hooks/assets.ts:61]: Do these two lines produce the same result as the code block above? 

https://github.com/pillarwallet/pillarwallet/pull/3160/files#diff-b026e55a5ec53394f0aa7f42bbf91191072902026723d551af04bd7b1f6d0287R37

**Review by IAmKio — CHANGES_REQUESTED** [review]
Great work, few questions and suggestions. Thanks!

File: src/actions/authActions.js [hunk 6]
```diff
@@ -397,13 +392,8 @@ export const logoutAction = () => {
     await dispatch(resetAppServicesAction());
```
[the remark this answers is not in the document]
Comment (hemadri-tank-bacancy) [reply src/actions/authActions.js:392]: Yes, `dispatch` commands in not need to await. but if we put await here, waiting for dispatch response after compiling next.

File: src/components/layout/ButtonGroup.tsx (hunk 3 above)
Comment (hemadri-tank-bacancy) [reply src/components/layout/ButtonGroup.tsx:41]: Not working, if items is null. here items is mandatory.

File: src/actions/authActions.js [hunk 7]
```diff
@@ -227,6 +220,8 @@ export const loginAction = (pin: ?string, privateKey: ?string, onLoginSuccess: ?
     const keyBasedAccount = findKeyBasedAccount(accounts);
     if (!keyBasedAccount) dispatch(addAccountAction(address, ACCOUNT_TYPES.KEY_BASED));
 
+    await dispatch(setStableTokens());
```
[the remark this answers is not in the document]
Comment (hemadri-tank-bacancy) [reply src/actions/authActions.js]: Here removed await

File: src/actions/etherspotActions.js [hunk 8]
```diff
@@ -268,18 +258,35 @@ export const reserveEtherspotEnsNameAction = (username: string) => {
...
+
+    if (!activeAccount) {
+      return;
+    }
+
+    const supportedChain = getSupportedChains(activeAccount);
+
+    const tokens = await Promise.all(supportedChain.map((chain) => etherspotService.getStableAssets(chain)));
+
+    const filterArr = [];
```
[the remark this answers is not in the document]
Comment (hemadri-tank-bacancy) [reply src/actions/etherspotActions.js]: Fixed

File: src/actions/onboardingActions.js [hunk 9]
```diff
@@ -410,6 +410,9 @@ export const setupAppServicesAction = (privateKey: ?string) => {
 
     logBreadcrumb('onboarding', 'setupAppServicesAction: dispatching loadRemoteConfigWithUserPropertiesAction');
     dispatch(loadRemoteConfigWithUserPropertiesAction());
+
+    logBreadcrumb('onboarding', 'setupAppServicesAction: dispatching setStableTokens');
+    await dispatch(setStableTokens());
```
[the remark this answers is not in the document]
Comment (hemadri-tank-bacancy) [reply src/actions/onboardingActions.js]: Here removed await

File: src/components/Wallet/WalletSelection.tsx (hunk 1 above)
Comment (hemadri-tank-bacancy) [reply src/components/Wallet/WalletSelection.tsx]: Fixed

File: src/components/Wallet/WalletSelection.tsx (hunk 2 above)
Comment (hemadri-tank-bacancy) [reply src/components/Wallet/WalletSelection.tsx]: Added here interface

File: src/hooks/assets.ts (hunk 4 above)
Comment (hemadri-tank-bacancy) [reply src/hooks/assets.ts]: Updated here some conditions.

File: src/hooks/assets.ts (hunk 5 above)
Comment (hemadri-tank-bacancy) [reply src/hooks/assets.ts]: Updated here some conditions.

File: src/components/layout/ButtonGroup.tsx (hunk 3 above)
Comment (hemadri-tank-bacancy) [reply src/components/layout/ButtonGroup.tsx:41]: FYI. Here items list is in hardcoded. like Tokens, Stables

**Comment by hemadri-tank-bacancy** [comment]
* Updated some points

  1. Set hight light  when selecting stables or tokens
  2. Set hardcoded token list for new user
  3. Press fold out button to navigate Assets
  

https://user-images.githubusercontent.com/78720269/203560535-52dfbfc5-547d-424a-8046-86974a03e610.mp4

**Comment by hemadri-tank-bacancy** [comment]
* For new user

https://user-images.githubusercontent.com/78720269/203561579-f0704e32-deae-4be0-a767-a1b006be95e8.mp4

**Comment by hemadri-tank-bacancy** [comment]
* Wallet open by default
* Show current assets price

Simulator Screen Shot - iPhone 13 Pro Max - 2022-11-24 at 11 15 24

File: src/components/layout/ButtonGroup.tsx (hunk 3 above)
Comment (IAmKio) [reply src/components/layout/ButtonGroup.tsx:41]: Got it, thanks!

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**